docs/src/main/sphinx/connector/iceberg.rst

```diff
@@ -785,6 +785,85 @@ To retrieve the information about the data files of the Iceberg table ``test_tab
 
 
 
+The output of the query has the following columns:
+
+.. list-table:: Files columns
+  :widths: 25, 30, 45
+  :header-rows: 1
+
+  * - Name
+    - Type
+    - Description
+  * - ``content``
+    - ``integer``
+    - Type of content stored in the file.
+      The supported content types in Iceberg are:
+
+      * ``DATA(0)``
+      * ``POSITION_DELETES(1)``
+      * ``EQUALITY_DELETES(2)``
+  * - ``file_path``
+    - ``varchar``
+    - The data file location
+  * - ``file_format``
+    - ``varchar``
+    - The format of the data file
+  * - ``record_count``
+    - ``bigint``
+    - The number of entries contained in the data file
+  * - ``file_size_in_bytes``
+    - ``bigint``
+    - The data file size
+  * - ``column_sizes``
+    - ``map(integer, bigint)``
+    - Mapping between the Iceberg column ID and its corresponding size in the file
+  * - ``value_counts``
+    - ``map(integer, bigint)``
+    - Mapping between the Iceberg column ID and its corresponding count of entries in the file
+  * - ``null_value_counts``
+    - ``map(integer, bigint)``
+    - Mapping between the Iceberg column ID and its corresponding count of ``NULL`` values in the file
+  * - ``nan_value_counts``
+    - ``map(integer, bigint)``
+    - Mapping between the Iceberg column ID and its corresponding count of non numerical values in the file
+  * - ``lower_bounds``
+    - ``map(integer, bigint)``
+    - Mapping between the Iceberg column ID and its corresponding lower bound in the file
+  * - ``upper_bounds``
+    - ``map(integer, bigint)``
+    - Mapping between the Iceberg column ID and its corresponding upper bound in the file
+  * - ``key_metadata``
+    - ``varbinary``
+    - Metadata about the encryption key used to encrypt this file, if applicable
+  * - ``split_offsets``
+    - ``array(bigint)``
+    - List of recommended split locations.
+  * - ``equality_ids``
+    - ``array(integer)``
+    - The set of field IDs used for equality comparison in equality delete files
+
+``$all_files`` table
+^^^^^^^^^^^^^^^^^^^^
+
+The ``$all_files`` table provides a detailed overview of the data files in all
+snapshots of the Iceberg table. The output of this metadata table can contain
+extra data files compare to the ``$files`` tables. One of the popssibility is when the table is rolled back to an older snapshot.
```

Comment:
> The ``$all_files`` table provides a detailed overview of the data files in all snapshots of the Iceberg table. The output of this metadata table can contain extra data files compare to the ``$files`` tables. One of the popssibility is when the table is rolled back to an older snapshot.

Let's replace it with the javadoc comment of `org.apache.iceberg.AllDataFilesTable` class:

> The `$all_files` table exposes the valid data files of the Iceberg table.
> A valid data file is one that is readable from any snapshot currently tracked by the table.